proto/api/v1/notification_service.proto

```diff
@@ -0,0 +1,117 @@
+syntax = "proto3";
+
+option go_package = "v1";
+
+option java_package = "io.stackrox.proto.api.v1";
+
+import weak "google/api/annotations.proto";
+import "api/v1/common.proto";
+import "api/v1/pagination.proto";
+import "google/protobuf/timestamp.proto";
+
+package v1;
+
+message NotificationsResponse {
+    repeated Notification notifications = 1;
+    Pagination              pagination  = 2;
+}
+
+message GetNotificationsRequest {
+    // For filtering notifications starting from a specific timestamp, i.e. the lower boundary.
+    google.protobuf.Timestamp from      = 1;
+
+    // For filtering notifications up until a specific timestamp, i.e. the upper boundary.
+    google.protobuf.Timestamp until     = 2;
+
+    // For filtering notifications from a specific area.
+    string area                         = 3;
+
+    // For filtering notifications associated with a specific resource type.
+    string resource_type                = 4;
+
+    // For filtering notifications based on their type.
+    NotificationType notification_type  = 5;
+
+    // For filtering notifications based on their level.
+    NotificationLevel level             = 6;
+}
+
+// Notifications are emitted by Central. They are used to create transparency for users
+// for asynchronous, background tasks.
+// Currently, notifications only cover logs emitted by Central.
+message Notification {
+    // UUID of the notification.
+    string id                               = 1;
```

Comment:
Since we talked about that today: Should we add a dedup key? Would be populated by the hash of other values and provide a performant lookup via an index column.

Reply:
Hmm my plan would have been to create a deterministic UUID based off the hash of the fields we want to use for deduping.

Not sure how supported it is to add a separate index column in combination with the generated store codes. Would rather keep this simple here.

We can also think about moving from UUID for the PK to a simple string and then use the hash as the PK. I'm fine with either option to be honest.

Reply:
> Hmm my plan would have been to create a deterministic UUID based off the hash of the fields we want to use for deduping.

Yeah that's what I meant, didn't formulate it precisely.

> Not sure how supported it is to add a separate index column in combination with the generated store codes. Would rather keep this simple here.

What do you mean? Wouldn't it just be a string field with a "hidden search" annotation? Or do you mean the dedup key should not be part of the notification proto itself, but a more general message that contains the notification as payload + the dedup key?

Reply:
Ah do you mean that `id` is already the dedup key?

Reply:
Well my plan would have been to just use the ID as the dedupe key, and subsequently as primary key instead of adding another field / message that holds the dedupe key. Since the dedupe key uniquely identifies an event.

Reply:
Yeah makes sense. Maybe clarify in the comment that it's not a normal UUIDv4 then.

Reply:
Added the comment (however, on the storage message, since otherwise it would be in the API doc and it's not really something users care about).